src/test/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/CoverageVisualisationServiceTest.kt

```diff
@@ -0,0 +1,189 @@
+    private val branch5 = Mockito.mock(BranchInfo::class.java)
+
+    private val mutation1 = Mockito.mock(MutationInfo::class.java)
+    private val mutation2 = Mockito.mock(MutationInfo::class.java)
+    private val mutation3 = Mockito.mock(MutationInfo::class.java)
+    private val mutation4 = Mockito.mock(MutationInfo::class.java)
+    private val mutation5 = Mockito.mock(MutationInfo::class.java)
+
+    @BeforeEach
+    fun setUp() {
+        val projectBuilder: TestFixtureBuilder<IdeaProjectTestFixture> =
+            IdeaTestFixtureFactory.getFixtureFactory().createFixtureBuilder("TestGenie")
+
+        fixture = JavaTestFixtureFactory.getFixtureFactory().createCodeInsightFixture(projectBuilder.fixture)
+        fixture.setUp()
+
+        project = fixture.project
+        coverageVisualisationService = project.service()
+        coverageToolWindowDisplayService = project.service()
+
+        // Initialise the toolWindow TestGenie
+        ToolWindowManager.getInstance(project)
+            .registerToolWindow(RegisterToolWindowTask("TestGenie", ToolWindowAnchor.RIGHT))
+
+        // Disable coverage visualisation
+        val settingsState = ApplicationManager.getApplication().getService(TestGenieSettingsService::class.java).state
+        settingsState.showCoverage = false
+    }
+
+    @AfterEach
+    fun tearDown() {
+        fixture.tearDown()
+    }
+
+    @ParameterizedTest
+    @MethodSource("valueGenerator")
+    fun fillToolWindowContents(
+        className: String,
+        coveredLinesSet: Set<Int>,
+        uncoveredLinesSet: Set<Int>,
+        coveredBranchesSet: Set<BranchInfo>,
+        uncoveredBranchesSet: Set<BranchInfo>,
+        coveredMutationSet: Set<MutationInfo>,
+        uncoveredMutationSet: Set<MutationInfo>,
+        lineCoverage: String,
+        branchCoverage: String,
+        mutationCoverage: String
+    ) {
+        val compactReport = CompactReport(TestGenerationResultImpl())
+        compactReport.UUT = className
+        compactReport.allCoveredLines = coveredLinesSet
+        compactReport.allUncoveredLines = uncoveredLinesSet
+        compactReport.allCoveredBranches = coveredBranchesSet
+        compactReport.allUncoveredBranches = uncoveredBranchesSet
+        compactReport.allCoveredMutation = coveredMutationSet
+        compactReport.allUncoveredMutation = uncoveredMutationSet
+
+        coverageVisualisationService.showCoverage(compactReport, editor)
+        assertThat(className).isEqualTo(coverageToolWindowDisplayService.data[0])
+        assertThat(lineCoverage).isEqualTo(coverageToolWindowDisplayService.data[1])
+        assertThat(branchCoverage).isEqualTo(coverageToolWindowDisplayService.data[2])
+        assertThat(mutationCoverage).isEqualTo(coverageToolWindowDisplayService.data[3])
+    }
+
+    @Test
+    fun createToolWindowTabTestSingleContent() {
+        coverageVisualisationService.showCoverage(CompactReport(TestGenerationResultImpl()), editor)
```

Comment:
I have a highlight "Not enough information to infer type variable T", I do not know what this is. The tests seems to pass. Looks fine.

Reply:
Hmm that is very strange. I believe that you have an old version, which does not support type inference that well?